### PR TITLE
docs(new): add TSDoc comments to the Frame class

### PR DESCRIPTION
**Pull request by kaycebasques**
@jackfranklin || @mathiasbynens please let me know if I'm on the right track

### CODE REVIEW

File: src/FrameManager.ts [hunk 1]
```diff
@@ -441,6 +515,13 @@ export class Frame {
     return this._mainWorld.evaluate<ReturnType>(pageFunction, ...args);
   }
 
+  /**
+   * Queries the frame for the CSS selector. If there's no such element within the frame, the method resolves to `null`.
```
Comment (kaycebasques) [inline src/FrameManager.ts:519]: Format `null` as inline code here?

File: src/FrameManager.ts [hunk 2]
```diff
@@ -360,6 +360,80 @@ export class FrameManager extends EventEmitter {
   }
 }
 
+/**
+ * At every point of time, page exposes its current frame tree via the [page.mainFrame()](#pagemainframe) and [frame.childFrames()](#framechildframes) methods.
```
Comment (kaycebasques) [inline src/FrameManager.ts:364]: Usage of `the` is a bit inconsistent. Here `page` is described without `the` in front of it. But on line 367 it's `the page`.

Comment (jackfranklin) [reply src/FrameManager.ts:364]: you'll need to swap the links for the `@link` syntax that TSDoc uses: https://api-extractor.com/pages/tsdoc/tag_link/

File: src/FrameManager.ts [hunk 3]
```diff
@@ -360,6 +360,80 @@ export class FrameManager extends EventEmitter {
...
+ *
+ * [Frame] object's lifecycle is controlled by three events, dispatched on the page object:
+ * - ['frameattached'](#event-frameattached) - fired when the frame gets attached to the page. A Frame can be attached to the page only once.
+ * - ['framenavigated'](#event-framenavigated) - fired when the frame commits navigation to a different URL.
+ * - ['framedetached'](#event-framedetached) - fired when the frame gets detached from the page.  A Frame can be detached from the page only once.
```
Comment (jackfranklin) [inline src/FrameManager.ts:369]: Once these are documented on the `Page` class we can link to them but we won't be able to yet - so maybe just make these:

. `framedetached` - fires when the ...

File: src/FrameManager.ts [hunk 4]
```diff
@@ -360,6 +360,80 @@ export class FrameManager extends EventEmitter {
...
+ * @example
+ * An example of getting text from an iframe element:
+ *
+ * ```js
+ *   const frame = page.frames().find(frame => frame.name() === 'myframe');
+ *   const text = await frame.$eval('.selector', element => element.textContent);
+ *   console.log(text);
+ * ```
+ *
+ * @public
```
Comment (jackfranklin) [inline src/FrameManager.ts:433]: this looks great - I think you'll want to use `@remarks` to split a summary from the full description: https://api-extractor.com/pages/tsdoc/tag_remarks/

File: src/FrameManager.ts (hunk 1 above)
Comment (jackfranklin) [reply src/FrameManager.ts:519]: 👍

**Comment by jackfranklin** [comment]
@kaycebasques definitely on the right track :) Thank you!

**Comment by jackfranklin** [comment]
@kaycebasques hey, do you have any time soon to finish this off? Doesn't matter at all, I'm very happy to do the final tidying up and merge it, but don't want to if you're planning to :)

**Comment by jackfranklin** [comment]
#6188 picks this up and adds docs for all the frame methods, building off this start. Thanks @kaycebasques :)

**Pull request closed without merging**